modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1227,9 +1311,30 @@ private CompletableFuture<Void> finishAndCleanup(
     ) {
         CompletableFuture<?> changeStateFuture = finishTransaction(enlistedPartitions, txId, commit, commitTimestamp, txCoordinatorId);
 
+        return cleanup(changeStateFuture, enlistedPartitions, commit, commitTimestamp, txId, ATTEMPTS_TO_CLEANUP_REPLICA)
+                .thenRun(() -> markLocksReleased(
+                        txId,
+                        enlistedPartitions,
+                        commit ? COMMITED : ABORTED,
+                        commitTimestamp)
+                );
+    }
+
+    private CompletableFuture<Void> cleanup(UUID txId, TxMeta txMeta) {
+        return cleanup(completedFuture(null), txMeta.enlistedPartitions(), txMeta.txState() == COMMITED, txMeta.commitTimestamp(), txId, 1);
```

Comment:
Why is this cleanup limited by attempts, but the durable cleanup is not? Might will we use the durable version here?

Reply:
This is limited because it is triggered from commit partition. There is a cleanup with retry with also limited count of attempts (see `ATTEMPTS_TO_CLEANUP_REPLICA`), but we agreed to remove the limit, we will need separate ticket for this.